### PR TITLE
Update upgrade-tools

### LINKED ISSUE #981: Can't update manager due to gnosis safe api url change
Upgrade script is working correctly, but can not connect to gnosis api due to recent gnosis safe api url change  
Potentially same issue are applicable to IMA  
Release candidates should be remade for correct upgrade of mainnet contracts  

Upgrade: `1.9.3-beta.4` -> `1.9.4-beta.0`
Log of manager upgrade:
 
 ```
 Generating typings for: 0 artifacts in dir: typechain-types for target: ethers-v5
Successfully generated 3 typings!
Successfully generated 7 typings for external artifacts!
Will mark updated version as 1.9.4-beta.0
Prepare upgrade of ContractManager
Contract ContractManager is up to date
Prepare upgrade of DelegationController
Contract DelegationController is up to date
Prepare upgrade of DelegationPeriodManager
Contract DelegationPeriodManager is up to date
Prepare upgrade of Distributor
Contract Distributor is up to date
Prepare upgrade of Punisher
Contract Punisher is up to date
Prepare upgrade of SlashingTable
Contract SlashingTable is up to date
Prepare upgrade of TimeHelpers
Contract TimeHelpers is up to date
Prepare upgrade of TokenState
Contract TokenState is up to date
Prepare upgrade of ValidatorService
Contract ValidatorService is up to date
Prepare upgrade of ConstantsHolder
Contract ConstantsHolder is up to date
Prepare upgrade of Nodes
Warning: Potentially unsafe deployment of Nodes

    You are using the `unsafeAllow.external-library-linking` flag to include external libraries.
    Make sure you have manually checked that the linked libraries are upgrade safe.

Nothing to compile
Generating typings for: 0 artifacts in dir: typechain-types for target: ethers-v5
Successfully generated 3 typings!
Successfully generated 7 typings for external artifacts!
Compiling 1 file with 0.8.17
Successfully submitted source code for contract
contracts/Nodes.sol:Nodes at 0xF9C383C95DE7D295343b7956477111C94Ed55809
for verification on the block explorer. Waiting for verification result...

Successfully verified contract Nodes on Etherscan.
https://goerli.etherscan.io/address/0xF9C383C95DE7D295343b7956477111C94Ed55809#code
Prepare upgrade of NodeRotation
Compiling 105 files with 0.8.17
Generating typings for: 120 artifacts in dir: typechain-types for target: ethers-v5
Successfully generated 199 typings!
Successfully generated 7 typings for external artifacts!
Solidity compilation finished successfully
Compiling 1 file with 0.8.17
Successfully submitted source code for contract
contracts/NodeRotation.sol:NodeRotation at 0x5FA723822B283955AD7028f4F8bff179e91a00d3
for verification on the block explorer. Waiting for verification result...

Successfully verified contract NodeRotation on Etherscan.
https://goerli.etherscan.io/address/0x5FA723822B283955AD7028f4F8bff179e91a00d3#code
Prepare upgrade of SchainsInternal
Contract SchainsInternal is up to date
Prepare upgrade of Schains
Contract Schains is up to date
Prepare upgrade of Decryption
Contract Decryption is up to date
Prepare upgrade of ECDH
Contract ECDH is up to date
Prepare upgrade of KeyStorage
Contract KeyStorage is up to date
Prepare upgrade of SkaleDKG
Warning: Potentially unsafe deployment of SkaleDKG

    You are using the `unsafeAllow.external-library-linking` flag to include external libraries.
    Make sure you have manually checked that the linked libraries are upgrade safe.

Compiling 105 files with 0.8.17
Generating typings for: 120 artifacts in dir: typechain-types for target: ethers-v5
Successfully generated 199 typings!
Successfully generated 7 typings for external artifacts!
Solidity compilation finished successfully
Compiling 1 file with 0.8.17
Successfully submitted source code for contract
contracts/SkaleDKG.sol:SkaleDKG at 0xF316989c412787Cdd9435883DC653EE3291f1831
for verification on the block explorer. Waiting for verification result...

Successfully verified contract SkaleDKG on Etherscan.
https://goerli.etherscan.io/address/0xF316989c412787Cdd9435883DC653EE3291f1831#code
Prepare upgrade of SkaleVerifier
Contract SkaleVerifier is up to date
Prepare upgrade of SkaleManager
Contract SkaleManager is up to date
Prepare upgrade of Pricing
Contract Pricing is up to date
Prepare upgrade of BountyV2
Contract BountyV2 is up to date
Prepare upgrade of Wallets
Contract Wallets is up to date
Prepare upgrade of SyncManager
Contract SyncManager is up to date
Prepare transaction to upgrade Nodes at 0x9E0A1C41E922b826A38E27eDb0359830c7B43F41 to 0xF9C383C95DE7D295343b7956477111C94Ed55809
Prepare transaction to upgrade NodeRotation at 0x9f90287f4cae4A51c6A0ed0e3e9d62f1cb4B7772 to 0x5FA723822B283955AD7028f4F8bff179e91a00d3
Prepare transaction to upgrade SkaleDKG at 0x69E2C9576B36c8d7938Ba45a1C909B67EC53048E to 0xF316989c412787Cdd9435883DC653EE3291f1831
Owner is a contract
Using Gnosis Safe
AxiosError: getaddrinfo ENOTFOUND safe-transaction.goerli.gnosis.io
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:71:26) {
  hostname: 'safe-transaction.goerli.gnosis.io',
  syscall: 'getaddrinfo',
  code: 'ENOTFOUND',
  errno: -3008,
  config: {
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    },
    adapter: [Function: httpAdapter],
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 0,
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    env: { FormData: [Function] },
    validateStatus: [Function: validateStatus],
    headers: {
      Accept: 'application/json, text/plain, */*',
      'User-Agent': 'axios/0.27.2'
    },
    method: 'get',
    url: 'https://safe-transaction.goerli.gnosis.io/api/v1/safes/0xeE4D4Af96296e556e376e844F7ED38c042a5F0F0/',
    data: undefined
  },
  request: <ref *1> Writable {
    _writableState: WritableState {
      objectMode: false,
      highWaterMark: 16384,
      finalCalled: false,
      needDrain: false,
      ending: false,
      ended: false,
      finished: false,
      destroyed: false,
      decodeStrings: true,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: true,
      bufferProcessing: false,
      onwrite: [Function: bound onwrite],
      writecb: null,
      writelen: 0,
      afterWriteTickInfo: null,
      buffered: [],
      bufferedIndex: 0,
      allBuffers: true,
      allNoop: true,
      pendingcb: 0,
      prefinished: false,
      errorEmitted: false,
      emitClose: true,
      autoDestroy: true,
      errored: null,
      closed: false
    },
    _events: [Object: null prototype] {
      response: [Function: handleResponse],
      error: [Function: handleRequestError],
      socket: [Function: handleRequestSocket]
    },
    _eventsCount: 3,
    _maxListeners: undefined,
    _options: {
      maxRedirects: 21,
      maxBodyLength: 10485760,
      protocol: 'https:',
      path: '/api/v1/safes/0xeE4D4Af96296e556e376e844F7ED38c042a5F0F0/',
      method: 'GET',
      headers: [Object],
      agent: undefined,
      agents: [Object],
      auth: undefined,
      hostname: 'safe-transaction.goerli.gnosis.io',
      port: null,
      nativeProtocols: [Object],
      pathname: '/api/v1/safes/0xeE4D4Af96296e556e376e844F7ED38c042a5F0F0/'
    },
    _ended: true,
    _ending: true,
    _redirectCount: 0,
    _redirects: [],
    _requestBodyLength: 0,
    _requestBodyBuffers: [],
    _onNativeResponse: [Function (anonymous)],
    _currentRequest: ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: false,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: false,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: 0,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      socket: [TLSSocket],
      _header: 'GET /api/v1/safes/0xeE4D4Af96296e556e376e844F7ED38c042a5F0F0/ HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'User-Agent: axios/0.27.2\r\n' +
        'Host: safe-transaction.goerli.gnosis.io\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: noopPendingOutput],
      agent: [Agent],
      socketPath: undefined,
      method: 'GET',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/api/v1/safes/0xeE4D4Af96296e556e376e844F7ED38c042a5F0F0/',
      _ended: false,
      res: null,
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'safe-transaction.goerli.gnosis.io',
      protocol: 'https:',
      _redirectable: [Circular *1],
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    _currentUrl: 'https://safe-transaction.goerli.gnosis.io/api/v1/safes/0xeE4D4Af96296e556e376e844F7ED38c042a5F0F0/',
    [Symbol(kCapture)]: false
  }
}
```

### LINKED ISSUE #1542: Can't update IMA due to gnosis safe api url change
```[tasklist]
### Tasks
- [ ] https://github.com/skalenetwork/IMA/issues/1638
```

### PR DESCRIPTION
**Pull request by DimaStebaev**

### CODE REVIEW

**Review by sergiy-skalelabs — APPROVED** [review]

**Review by DmytroNazarenko — APPROVED** [review]

File: proxy/migrations/deploySchain.ts [hunk 1]
```diff
@@ -52,14 +52,14 @@ export function getContractKeyInAbiFile(contract: string): string {
...
 
 export async function getManifestFile(): Promise<string> {
-    return (await Manifest.forNetwork(ethers.provider)).file;;
+    return (await Manifest.forNetwork(ethers.provider)).file;
 }
 
-export function getProxyMainnet(contractName: string) {
-    const defaultFilePath = "../data/proxyMainnet.json";
-    const jsonData = require(defaultFilePath);
+async function getProxyMainnet(contractName: string) {
```
Comment (payvint) [inline proxy/migrations/deploySchain.ts:58]: async??

File: proxy/migrations/deploySchain.ts [hunk 2]
```diff
@@ -194,8 +194,38 @@ async function main() {
...
+    during deployment of smart contract
+    that exceed 24KB limit.
+
+    In addition to this problem
+    upgrade-hardhat library
+    did not supported
+    manual gas limit configuration.
+
+    TODO: in case of any one or both issues fixed
+    please remove this crazy workaround below
```
Comment (payvint) [inline proxy/migrations/deploySchain.ts:211]: Thank you!!!
It become much more interesting to read this PR:)

File: proxy/migrations/deploySkaleManagerComponents.ts [hunk 3]
```diff
@@ -156,7 +156,7 @@ async function main() {
     };
     await keyStorage.setBlsCommonPublicKeyForSchain( ethers.utils.solidityKeccak256(['string'], [schainName]), BLSPublicKey );
     console.log("Set common public key in KeyStorage contract", keyStorage.address, "\n");
-    await wallets.rechargeSchainWallet( web3.utils.soliditySha3( schainName ), { value: "10000000000000000000" } ); // originally it was 1000000000000000000 = 1ETH
+    await wallets.rechargeSchainWallet( ethers.utils.solidityKeccak256( ["string"], [schainName] ), { value: "10000000000000000000" } ); // originally it was 1000000000000000000 = 1ETH
```
Comment (payvint) [inline proxy/migrations/deploySkaleManagerComponents.ts]: Maybe using stringKeccak256 from helper?

File: proxy/test/DepositBoxEth.ts [hunk 4]
```diff
@@ -28,40 +28,32 @@ import chaiAsPromised from "chai-as-promised";
...
-    DepositBoxERC721,
-    DepositBoxERC1155,
     Linker,
     MessageProxyForMainnet,
     MessagesTester,
-    ERC20OnChain,
     CommunityPool
 } from "../typechain";
-import { stringFromHex, stringValue, getPublicKey } from "./utils/helper";
+import { stringKeccak256, getBalance, getPublicKey } from "./utils/helper";
```
Comment (payvint) [inline proxy/test/DepositBoxEth.ts:36]: Is getBalance used somewhere?

File: proxy/test/TokenManagerERC20.ts [hunk 5]
```diff
@@ -64,7 +64,7 @@ describe("TokenManagerERC20", () => {
     const schainId = ethers.utils.solidityKeccak256(["string"], [schainName]);
```
Comment (payvint) [inline proxy/test/TokenManagerERC20.ts]: Use solidityKeccak256 from helper?

**Review by payvint — COMMENTED** [review]
Looks solid

File: proxy/migrations/deploySchain.ts (hunk 1 above)
Comment (DimaStebaev) [reply proxy/migrations/deploySchain.ts:58]: Yes. `readFile` is an async call.

File: proxy/test/DepositBoxEth.ts (hunk 4 above)
Comment (DimaStebaev) [reply proxy/test/DepositBoxEth.ts:36]: Yes.
At line # 76 for example.

**Comment by DimaStebaev** [comment]
> Looks solid

ity.

**Review by payvint — APPROVED** [review]

**Pull request merged**